plugins/algoltek-usb/fu-algoltek-usb-firmware.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2023 Ling.Chen <ling.chen@algoltek.com.tw>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include "fu-algoltek-usb-firmware.h"
+#include "fu-algoltek-usb-struct.h"
+
+struct _FuAlgoltekUsbFirmware {
+    FuFirmware parent_instance;
+};
+
+G_DEFINE_TYPE(FuAlgoltekUsbFirmware, fu_algoltek_usb_firmware, FU_TYPE_FIRMWARE)
+
+#define FU_ALGOLTEK_HEADER 0x4B45544C4F474C41 // ALGOLTEK
+
+#define FU_ALGOLTEK_FIRMWARE_SIZE 0x20000
+#define FU_ALGOLTEK_ISP_SIZE      0x1000
+
+static gboolean fu_algoltek_usb_check_magic(FuFirmware *firmware, GBytes *fw, gsize offset,
```

Comment:
This vfunc is changed in main branch, can you please rebase?

Reply:
I changed the function name from "check_magic" to "validate".